### ISSUE TITLE
Does exoplayer automatically use ABR as per internet connectivity?

### ISSUE THREAD

**Issue by kvmoondra** [google/ExoPlayer#7703]
Hello there,
Hope you are well.

With regards, we are developing a video news android app(from our app: we are going to distribute the videos to end-users, only video content).

**For the same, we are looking to integrate Exoplayer(as a video player to stream the content) but we have a query:**

-Does Exoplayer automatically use ABR and stream as per Internet connectivity? What I mean over here is: normal MP4 links from AWS/google cloud to the player, no M3U8 links will be provided by AWS/Google cloud to the player & player streams as per internet connectivity of the end-user without getting M3U8 link from AWS )? Does this work? OR 

-Does Exoplayer needs the inputs(in the form of M3U8) from Cloud(like AWS or Google cloud) and then only Exoplayer can stream HLS?

Will wait to hear from you

Thanks & Regards
Kumar Vansh Moondra

**Comment by ojw28** [comment]
If you play adaptive media (i.e., DASH, HLS or SmoothStreaming), then ExoPlayer will do ABR by default. If you play progressive media (i.e., regular media files such as MP4s), then ExoPlayer will not do ABR.

Given a progressive media file, there isn't really a way that a player could do ABR even if it wanted to. A player could theoretically do ABR if provided with multiple progressive media files at different bitrates, but that's exactly what adaptive streaming standards such as DASH, HLS and SmoothStreaming are designed to do. So that's what you should use.

In conclusion, if you want ExoPlayer to do ABR then you need to use an adaptive streaming standard (i.e., DASH, HLS or SmoothStreaming) for your content.

**Comment by kvmoondra** [comment]
Thanks, @ojw28 for your reply. 

**Regarding HLS streaming**, you meant to say: If one wants ABR through Exoplayer then we need to give .M3U8 input then only Exoplayer can stream ABR? Player won't use ABR technology(HLS) if we give.MP4 input?

Correct me If I am wrong

Thanks & Regards
Kumar Vansh Moondra

**Comment by ojw28** [comment]
Correct.
